src/runtime/client-event-manager-test.js

```diff
@@ -34,272 +35,310 @@ const DEFAULT_EVENT = {
 };
 
 describes.sandboxed('EventManager', {}, () => {
-  describe('promises', () => {
-    let eventMan;
-    let resolver;
-    let rejector;
-    let counter;
+  describe('configuration', () => {
+    let eventManager;
+    let indicateConfigurationSucceeded;
+    let indicateConfigurationFailed;
+    let events;
+    let configurationPromise;
+
     beforeEach(() => {
-      eventMan = new ClientEventManager(
-        new Promise((resolve, reject) => {
-          rejector = reject;
-          resolver = resolve;
-        })
-      );
-      counter = 0;
-      eventMan.registerEventListener(() => counter++);
-      eventMan.logEvent(DEFAULT_EVENT);
-      //ensure it has not logged yet
-      expect(counter).to.equal(0);
+      configurationPromise = new Promise((resolve, reject) => {
+        indicateConfigurationSucceeded = resolve;
+        indicateConfigurationFailed = reject;
+      });
+      eventManager = new ClientEventManager(configurationPromise);
+      events = [];
+      eventManager.registerEventListener(e => events.push(e));
+    });
+
+    it('should not log events before configuration succeeds', async () => {
+      // Request event log.
+      eventManager.logEvent(DEFAULT_EVENT);
+
+      // Event should not be logged.
+      await tick();
+      expect(events).to.deep.equal([]);
     });
 
-    it('should not log events until its promise is resolved', async () => {
-      let counter2 = 0;
+    it('should log events after configuration succeeds', async () => {
+      // Request event log.
+      eventManager.logEvent(DEFAULT_EVENT);
+
+      indicateConfigurationSucceeded();
 
-      eventMan.registerEventListener(() => counter2++);
-      eventMan.logEvent(DEFAULT_EVENT);
-      //ensure it has not logged yet
-      expect(counter).to.equal(0);
-      expect(counter2).to.equal(0);
+      // Event should be logged.
+      await tick();
+      expect(events).to.deep.equal([DEFAULT_EVENT]);
+    });
 
-      resolver();
-      await eventMan.lastAction_;
-      //ensure it logged both events after we called resolver and yielded
-      expect(counter).to.equal(counter2);
+    it('should not log events if configuration fails', async () => {
+      // Request event log.
+      eventManager.logEvent(DEFAULT_EVENT);
 
-      //If this test is flaky it means sometimes event manager is logging
-      //despite the promise not being resolved (which is a problem).
+      indicateConfigurationFailed();
+
+      // Event should not be logged.
+      await tick();
+      expect(events).to.deep.equal([]);
     });
 
-    it('should not log events if promise rejected', async () => {
-      rejector();
-      try {
-        await eventMan.lastAction_;
-      } catch (err) {}
-      expect(counter).to.equal(0);
+    it('returns ready promise', () => {
+      expect(eventManager.getReadyPromise()).to.equal(configurationPromise);
```

Comment:
It surprises me that this test passes.  The configuration promise (passed in by the caller) is different from the ready promise (awaiting iframe load).

Reply:
In `client-event-manager.js` I'm seeing `this.isReadyPromise_ = configuredPromise;`